lib/sql_footprint.rb

```diff
@@ -34,9 +34,11 @@ def exclude
       @capture = true
     end
 
-    def capture sql
-      return unless @capture && @filter.capture?(sql)
-      @statements.add @anonymizer.anonymize(sql)
+    private
+
+    def capturer database_name
```

Comment:
Then you could implement `capturer` like so:
```ruby
def capturer database_name
  capturers[database_name]
end
```

Note that you don't want to use `fetch` here to take advantage of the default value block.